### ISSUE TITLE
update_link(): AttributeError: 'Page' object has no attribute 'super'

### ISSUE THREAD

**Issue by NtSalazar** [pymupdf/PyMuPDF#3070]
### Description of the bug

Hi!, i have an error when i try to modify an url in a pdf. the error disappear when you remove the ".super()"

## Traceback

This is the complete traceback I see:
```
Traceback (most recent call last):
  File "test.py", line 328, in <module>
    Send_Accessing_Message(user, response_message)
  File "test.py", line 240, in Send_Accessing_Message
    pdf[0].update_link(links[0])
  File "C:\Users\computer_name\AppData\Local\pypoetry\Cache\virtualenvs\main-S6_UyEDV-py3.8\lib\site-packages\fitz\utils.py", line 1716, in update_link
    page.parent.update_object(lnk["xref"], annot, page=page)
  File "C:\Users\computer_name\AppData\Local\pypoetry\Cache\virtualenvs\main-S6_UyEDV-py3.8\lib\site-packages\fitz\__init__.py", line 5615, in update_object
    JM_refresh_links( mupdf.pdf_page_from_fz_page(page.super()))
AttributeError: 'Page' object has no attribute 'super'
```

### How to reproduce the bug

```cmd
poetry run python test.py
```

test.py
```python
import fitz
# update the link
with fitz.open("modify-url.pdf") as pdf:
  links = pdf[0].get_links()
  links[0]['uri'] = "https://www.ddg.gg"
  pdf[0].update_link(links[0])
  pdf.save("output.pdf")
```
[modify-url.pdf](https://github.com/pymupdf/PyMuPDF/files/14002768/modify-url.pdf)

### PyMuPDF version

1.23.16

### Operating system

Windows

### Python version

3.12

**Comment by julian-smith-artifex-com** [comment]
Thanks for reporting this.

I have a fix in my tree which we'll hopefully release in the next day or two.

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.23.17.
